tests/contrib/funsor/test_vectorized_markov.py

```diff
@@ -0,0 +1,187 @@
+    y_probs = pyro.param("y_probs", lambda: torch.rand(w_dim, x_dim, y_dim), constraint=constraints.simplex)
+
+    w_prev = x_prev = None
+    markov_loop = \
+        pyro.vectorized_markov(name="time", size=len(data), dim=-2, history=history) if vectorized \
+        else pyro.markov(range(len(data)), history=history)
+    for i in markov_loop:
+        w_curr = pyro.sample(
+            "w_{}".format(i), dist.Categorical(
+                w_init if isinstance(i, int) and i < 1 else w_trans[w_prev]),
+            infer={"enumerate": "parallel"})
+        x_curr = pyro.sample(
+            "x_{}".format(i), dist.Categorical(
+                x_init[w_curr] if isinstance(i, int) and i < 1 else x_trans[w_curr, x_prev]),
+            infer={"enumerate": "parallel"})
+        with pyro.plate("tones", data.shape[-1], dim=-1):
+            pyro.sample("y_{}".format(i), dist.Categorical(
+                Vindex(y_probs)[w_curr, x_curr]),
+                obs=data[i])
+        x_prev, w_prev = x_curr, w_curr
+
+
+@pytest.mark.parametrize("model,data,var", [
+    (model_0, torch.rand(5), "xy"),
+    (model_1, torch.rand(5, 4), "xy"),
+    (model_2, torch.ones((5, 4), dtype=torch.long), "xy"),
+    (model_3, torch.ones((5, 4), dtype=torch.long), "wxy"),
+    (model_4, torch.ones((5, 4), dtype=torch.long), "wxy"),
+])
+def test_vectorized_markov(model, data, var):
+    history = 1
+
+    with pyro_backend("contrib.funsor"), \
+            handlers.enum():
+        trace = handlers.trace(model).get_trace(data, False)
+        factors = list()
+        for i in range(len(data)):
+            for v in var:
+                factors.append(trace.nodes["{}_{}".format(v, i)]["funsor"]["log_prob"])
+
+        vectorized_trace = handlers.trace(model).get_trace(data, True)
+        vectorized_factors = list()
+        for i in range(history):
+            for v in var:
+                vectorized_factors.append(vectorized_trace.nodes["{}_{}".format(v, i)]["funsor"]["log_prob"])
+        for i in range(history, len(data)):
+            for v in var:
+                vectorized_factors.append(
+                    vectorized_trace.nodes["{}_{}".format(v, torch.arange(history, len(data)))]["funsor"]["log_prob"]
+                    (**{"time": i-history,
+                        "x_{}".format(torch.arange(history, len(data))): "x_{}".format(i),
+                        "x_{}".format(torch.arange(len(data)-history)): "x_{}".format(i-history),
+                        "w_{}".format(torch.arange(history, len(data))): "w_{}".format(i),
+                        "w_{}".format(torch.arange(len(data)-history)): "w_{}".format(i-history),
+                        "y_{}".format(torch.arange(history, len(data))): "y_{}".format(i)}),
+                    )
+
+        for f1, f2 in zip(factors, vectorized_factors):
+            assert set(f1.inputs) == set(f2.inputs)
+            assert torch.equal(pyro.to_data(f1), pyro.to_data(f2))
```

Comment:
It checks the `step` information now. Note: instead of the `plate_to_step` there is a `time` node in the trace with `step` information in the `infer` dict.